test/client/fixtures/sandbox/selection-test.js

```diff
@@ -0,0 +1,26 @@
+var selectionSandbox = hammerhead.eventSandbox.selection;
+var browserUtils     = hammerhead.utils.browser;
+
+var inputTestValue = 'test@host.net';
+var isIE           = browserUtils.isIE;
+
+test('Set and get selection on input with "email" type', function () {
+    var input = document.createElement('input');
+
+    input.type  = 'email';
+    input.value = inputTestValue;
+
+    document.body.appendChild(input);
+
+    input.setSelectionRange(0, inputTestValue.length, 'backward');
```

Comment:
use some custom selection in this test (not form start to end, but something like 1..3)